### DISCUSSION TITLE
csrfguard.js correct way to config

### DISCUSSION THREAD

**Discussion by Sabrinale** [OWASP/www-project-csrfguard#194] [Q&A] [answered]
Hello,
I have set up latest version 4.2. and added csrfGuard.properties in webapp/WEB-INF. However, the app work without out csrfguard.js and in the of every page does not seem to have the csrfGuard.js file (when i  inspect element). When this js file needed. I currently need to inject token name and token value every single form

( ex <form id="login_form" action="<%= request.getContextPath() %>/servlet/StandardServlet?action=login" method="post">
            <input type="hidden" name="<csrf:tokenname/>" value="<csrf:tokenvalue/>"/>

it works and token showed in views I added . in web.xml has sth like "inject in attribute, inject-into-forms" 
<init-param>
            <param-name>inject-into-forms</param-name>
            <param-value>true</param-value>
        </init-param>
        <init-param>
            <param-name>inject-into-attributes</param-name>
            <param-value>true</param-value>
So i wonder if i installed correctly. Do I need to add token name and token value manually to form. Thank you for helping. in the csrfguard,js mention "  Code to ensure our event always gets triggered when the DOM is updated or ie does not properly support prototype - wrap completely" . how can i test the application so the break point would go to this csrfguards.js file?

**Comment by forgedhallpass** [comment, marked as answer]
The JavaScript is needed in every case, except when you have a JSP application, because in that case the tokens are injected directly on the server-side. 

Duplicate of #193
